File: Data Structures & Algorithms/remove-element/submission-0.cs
public class Solution {
    public int RemoveElement(int[] nums, int val) 
    {
        int non_val_index = 0;
        int i = 0;

        while(non_val_index < nums.Length && i < nums.Length)
        {
            int current_non_val = nums[non_val_index];

            if (current_non_val == val)
            {
                non_val_index++;
                continue;
            }
            else
            {
                nums[i] = current_non_val;
                i++;
                non_val_index++;
            }
        }
        return i;

    }
}

File: Data Structures & Algorithms/contains-duplicate-ii/submission-0.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        for i in range(len(nums)):
            # j is <= k away from i
            lim = min(i + k, len(nums) - 1) 
            j = i + 1

            while j <= lim:
                if nums[i] == nums[j]:
                    return True
                j = j + 1

        return False